src/main/java/com/google/firebase/auth/UserRecord.java

```diff
@@ -528,6 +532,17 @@ UpdateRequest setValidSince(long epochSeconds) {
       return this;
     }
 
+    UpdateRequest linkProvider(@NonNull UserProvider userProvider) {
+      properties.put("linkProviderUserInfo", userProvider);
+      return this;
+    }
+
+    UpdateRequest deleteProvider(String providerId) {
+      checkProviderId(providerId);
+      properties.put("deleteProvider", ImmutableList.of(providerId));
```

Comment:
This would overwrite previous values such that:
```
UpdateRequest ur = ...;
ur.deleteProvider("google.com").deleteProvider("facebook.com")
```
would result in only the facebook.com provider being deleted.

Better might be: `UpdateRequest setDeleteProviders(Iterable<String> providerIds)`. You could optionally add `UpdateRequest addDeleteProvider(String)` and/or `UpdateRequest addDeleteProviders(Iterable<String>)`.

See further discussion here: http://go/java-practices/builders#special. (Second bullet point).